### PR TITLE
Implement table comments

### LINKED ISSUE #23225: how to set table comment
I can not set table comment in clickhouse like in myql:
```
ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
COMMENT '标签项血缘依赖表';
```

Is there any document about table comment syntax, I can not find any by myself. thx.

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Implement table comments. closes #23225.

### CODE REVIEW

File: tests/queries/0_stateless/01821_table_comment.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+)
+ENGINE = Log
+COMMENT 'this is a Log table';
+
+SELECT
+    name,
+    comment
+FROM system.tables
+WHERE name IN ('t1', 't2', 't3') order by name;
+
```
Comment (Enmk) [inline tests/queries/0_stateless/01821_table_comment.sql:32]: Could you please add `SHOW CREATE TABLE t1` to the test?

**Comment by kssenii** [comment]
btw, `Functional stateless tests flaky check (address) — fail: 93, passed: 7` fails because need to add it to list of tests, which cannot be run in parallel (https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/skip_list.json#L404).

**Comment by ucasFL** [comment]
> btw, `Functional stateless tests flaky check (address) — fail: 93, passed: 7` fails because need to add it to list of tests, which cannot be run in parallel (https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/skip_list.json#L404).

Ok!

File: src/Storages/StorageInMemoryMetadata.h [hunk 2]
```diff
@@ -52,6 +54,8 @@ struct StorageInMemoryMetadata
     /// structure from different threads. It should be used as MultiVersion
     /// object. See example in IStorage.
 
+    void setComment(const String & comment_);
+
     /// Sets only real columns, possibly overwrites virtual ones.
     void setColumns(ColumnsDescription columns_);
```
Comment (kssenii) [inline src/Storages/StorageInMemoryMetadata.h:60]: May be add a comment like `Sets a user-defined comment for a table`

File: src/Storages/System/StorageSystemTables.cpp [hunk 3]
```diff
@@ -244,6 +244,9 @@ class TablesBlockSource : public SourceWithProgress
                         // lifetime_bytes
                         if (columns_mask[src_index++])
                             res_columns[res_index++]->insertDefault();
+
+                        if (columns_mask[src_index++])
+                            res_columns[res_index++]->insertDefault();
```
Comment (kssenii) [inline src/Storages/System/StorageSystemTables.cpp:249]: A comment needed, all such `if`'s above have a comment for inserted column name, this one needs too.

File: src/Storages/System/StorageSystemTables.cpp [hunk 4]
```diff
@@ -469,6 +472,12 @@ class TablesBlockSource : public SourceWithProgress
                     else
                         res_columns[res_index++]->insertDefault();
                 }
+
+                if (columns_mask[src_index++])
+                {
+                    assert(metadata_snapshot != nullptr);
+                    res_columns[res_index++]->insert(metadata_snapshot->comment);
+                }
```
Comment (kssenii) [inline src/Storages/System/StorageSystemTables.cpp:480]:
```suggestion

                if (columns_mask[src_index++])
                {
                    if (metadata_snapshot)
                        res_columns[res_index++]->insert(metadata_snapshot->comment);
                    else
                        res_columns[res_index++]->insertDefault();
                }
```

File: src/Storages/StorageInMemoryMetadata.h (hunk 2 above)
Comment (ucasFL) [reply src/Storages/StorageInMemoryMetadata.h:60]: Ok.

File: src/Parsers/ASTCreateQuery.cpp [hunk 5]
```diff
@@ -32,6 +32,11 @@ ASTPtr ASTStorage::clone() const
     if (settings)
         res->set(res->settings, settings->clone());
 
+    if (comment)
+    {
+        res->comment = comment->clone();
+    }
```
Comment (kssenii) [inline src/Parsers/ASTCreateQuery.cpp:38]: May be use `set` method as in the cases below?

**Review by kssenii — APPROVED** [review]

File: src/Parsers/ASTCreateQuery.cpp (hunk 5 above)
Comment (ucasFL) [reply src/Parsers/ASTCreateQuery.cpp:38]: Ok.

File: src/Parsers/ASTCreateQuery.h [hunk 6]
```diff
@@ -25,6 +25,8 @@ class ASTStorage : public IAST
     IAST * ttl_table = nullptr;
     ASTSetQuery * settings = nullptr;
 
+    ASTPtr comment;
```
Comment (kssenii) [inline src/Parsers/ASTCreateQuery.h:28]:
```suggestion
    IAST * ttl_table = nullptr;
    IAST * comment = nullptr;
    ASTSetQuery * settings = nullptr;
```
Why is it not like this? all other such values are just raw pointers `IAST *`, and this one is no different from them.

File: src/Parsers/ASTCreateQuery.cpp (hunk 5 above)
Comment (kssenii) [reply src/Parsers/ASTCreateQuery.cpp:38]: They differ only in type of pointer.

File: src/Parsers/ASTCreateQuery.h (hunk 6 above)
Comment (ucasFL) [reply src/Parsers/ASTCreateQuery.h]: OK! updated.

**Review by kssenii — APPROVED** [review]

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-9474.

**Pull request merged**